ui/jobs/jobs.ts

```diff
@@ -1002,31 +1089,32 @@ class Bars {
-    logs.push(t + '1A:10000000:' + this.me + ' gains the effect of Medicated from ' + this.me + ' for 30.2 Seconds.');
-    logs.push(t + '15:10000000:Tako Yaki:1D60:Embolden:10000000:' + this.me + ':500020F:4D70000:0:0:0:0:0:0:0:0:0:0:0:0:0:0:42194:42194:10000:10000:0:1000:-655.3301:-838.5481:29.80905:0.523459:42194:42194:10000:10000:0:1000:-655.3301:-838.5481:29.80905:0.523459:00001DE7');
-    logs.push(t + '1A:10000000:' + this.me + ' gains the effect of Battle Litany from  for 25 Seconds.');
-    logs.push(t + '1A:10000000:' + this.me + ' gains the effect of The Balance from  for 12 Seconds.');
+    const me = this.me ?? '';
+    logs.push(`${t}1A:10000000:${me} gains the effect of Medicated from ${me} for 30.2 Seconds.`);
+    logs.push(`${t}15:10000000:Tako Yaki:1D60:Embolden:10000000:${me}:500020F:4D70000:0:0:0:0:0:0:0:0:0:0:0:0:0:0:42194:42194:10000:10000:0:1000:-655.3301:-838.5481:29.80905:0.523459:42194:42194:10000:10000:0:1000:-655.3301:-838.5481:29.80905:0.523459:00001DE7'`);
+    logs.push(t + '1A:10000000:' + me + ' gains the effect of Battle Litany from  for 25 Seconds.');
+    logs.push(t + '1A:10000000:' + me + ' gains the effect of The Balance from  for 12 Seconds.');
     logs.push(t + '1A:10000000:Okonomi Yaki gains the effect of Foe Requiem from Okonomi Yaki for 9999.00 Seconds.');
     logs.push(t + '15:1048638C:Okonomi Yaki:8D2:Trick Attack:40000C96:Striking Dummy:20710103:154B:');
-    logs.push(t + '1A:10000000:' + this.me + ' gains the effect of Left Eye from That Guy for 15.0 Seconds.');
-    logs.push(t + '1A:10000000:' + this.me + ' gains the effect of Right Eye from That Guy for 15.0 Seconds.');
+    logs.push(t + '1A:10000000:' + me + ' gains the effect of Left Eye from That Guy for 15.0 Seconds.');
+    logs.push(t + '1A:10000000:' + me + ' gains the effect of Right Eye from That Guy for 15.0 Seconds.');
     logs.push(t + '15:1048638C:Tako Yaki:1D0C:Chain Stratagem:40000C96:Striking Dummy:28710103:154B:');
     logs.push(t + '15:1048638C:Tako Yaki:B45:Hypercharge:40000C96:Striking Dummy:28710103:154B:');
-    logs.push(t + '1A:10000000:' + this.me + ' gains the effect of Devotion from That Guy for 15.0 Seconds.');
-    logs.push(t + '1A:10000000:' + this.me + ' gains the effect of Brotherhood from That Guy for 15.0 Seconds.');
-    logs.push(t + '1A:10000000:' + this.me + ' gains the effect of Brotherhood from Other Guy for 15.0 Seconds.');
-    const e = { detail: { logs: logs } };
+    logs.push(t + '1A:10000000:' + me + ' gains the effect of Devotion from That Guy for 15.0 Seconds.');
+    logs.push(t + '1A:10000000:' + me + ' gains the effect of Brotherhood from That Guy for 15.0 Seconds.');
+    logs.push(t + '1A:10000000:' + me + ' gains the effect of Brotherhood from Other Guy for 15.0 Seconds.');
+    const e = { type: 'onLogEvent' as const, detail: { logs: logs } };
     this._onLogEvent(e);
   }
 }
 
-let gBars;
+let gBars: Bars;
 
 UserConfig.getUserConfigLocation('jobs', Options, () => {
```

Comment:
@quisquous I am not very understand how was this function working, can you help me, basically for typing the argument `Options`?

Reply:
My thought was that `const Options` up at the top of the file will become `type JobsOptions extends BaseOptions`.  And then this type here is `JobsOptions`.